libs/server/Resp/ArrayCommands.cs

```diff
@@ -668,15 +668,25 @@ private bool NetworkSELECT(byte* ptr)
 
             readHead = (int)(ptr - recvBufferPtr);
 
-            if (string.Equals(result, "0"))
+            if (storeWrapper.serverOptions.EnableCluster)
             {
-                while (!RespWriteUtils.WriteDirect(CmdStrings.RESP_OK, ref dcurr, dend))
+                // Cluster mode does not allow DBID
+                while (!RespWriteUtils.WriteError(CmdStrings.RESP_ERR_GENERIC_SELECT_CLUSTER_MODE, ref dcurr, dend))
                     SendAndReset();
             }
             else
             {
-                while (!RespWriteUtils.WriteError("ERR invalid database index."u8, ref dcurr, dend))
-                    SendAndReset();
+
+                if (!storeWrapper.serverOptions.EnableCluster && string.Equals(result, "0"))
```

Comment:
why do we need to check for EnableCluster here?

Reply:
Currently Redis does not allow for multiple databases in Cluster mode. While awaiting for the work to add support for this, I thought it would be good to at least inform clients with an error message that for cluster mode this feature is not available.

Reply:
Good catch, forgot to remove the second check. Removed it now!